builders/ruby.go

```diff
@@ -148,7 +152,34 @@ func (builder *RubyBuilder) IsModule(target string) (bool, error) {
 	return false, errors.New("IsModule is not implemented for RubyBuilder")
 }
 
-// InferModule is not implemented
-func (builder *RubyBuilder) InferModule(target string) (module.Module, error) {
-	return module.Module{}, errors.New("InferModule is not implemented for RubyBuilder")
+// DiscoverModules returns ModuleConfigs that match Gemfiles in the directory
+func (builder *RubyBuilder) DiscoverModules(dir string) ([]config.ModuleConfig, error) {
+	gemFilePaths, err := doublestar.Glob(filepath.Join(dir, "**", "Gemfile"))
```

Comment:
Should we be matching all Gemfiles?

Reply:
cc @lackstein, who has the most Ruby experience. I'm not sure this method of discovering Ruby modules is idiomatic.

Reply:
Pretty much every Ruby project will end up having its own `Gemfile`, so this is probably an okay way of finding them throughout the directory tree. But if they've got vendored dependencies, then those likely also have Gemfiles, and you'd be picking them up as well.

Reply:
That's fine, we will handle vendored deps separately in #52 via config, and this is desired behavior.

Reply:
Bundler puts vendorised gems into `vendor` by default. We should filter those out, unless `includeAll` is specified.